### PR TITLE
Education Profile

### PR DESCRIPTION
**Pull request by aidanouckama**
- Added implementation for user to add education / school
- Utilized an API from a GitHub repo: https://github.com/Hipo/university-domains-list
- Slightly changed the profile schema to include school
- Created a new search query for education

### CODE REVIEW

File: client/src/components/SchoolQuerySearch.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      // Remove duplicates based on the "name" property
+      const uniqueResults = data.reduce((unique, school) => {
+        if (!unique.find((item) => item.name === school.name)) {
+          unique.push(school);
+        }
+        return unique;
+      }, []);
+
+      console.log("data:", data);
+      console.log("query:", query);
```
Comment (Lewin-B) [inline client/src/components/SchoolQuerySearch.js]: Should remove these console logs before pushing to prod

File: client/src/components/SchoolQuerySearch.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            results.map((school) => (
+              <div
+                key={`${school.name}_result`}
+                className="flex flex-row justify-center items-center px-5 py-2 hover:bg-gray-100"
+              >
+                {/* <img
+                  className="w-10 h-10 rounded-lg object-contain"
+                  src={}
+                  alt={`school_`}
+                /> */}
```
Comment (Lewin-B) [inline client/src/components/SchoolQuerySearch.js]: Would remove unused code if not needed

**Review by Lewin-B — CHANGES_REQUESTED** [review]
Great work looks good for the most part just a few little things

File: client/src/components/SchoolQuerySearch.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  // initialize school if one exists
+  useEffect(() => {
+    // reset query before checking if theres a previous value
+    setQuery("");
+
+    // check if there was a previous value
+    // if there was, set query to it
+    if (value) {
+      setQuery(value);
+    }
```
Comment (arfpog) [inline client/src/components/SchoolQuerySearch.js]: just do:

value ? setQuery(value) : setQuery("")

File: client/src/components/SchoolQuerySearch.js (hunk 1 above)
Comment (arfpog) [reply client/src/components/SchoolQuerySearch.js]: ^

File: client/src/pages/EditProfile.js [hunk 4]
```diff
@@ -47,6 +51,7 @@ function EditProfile() {
       .then((data) => {
```
Comment (arfpog) [inline client/src/pages/EditProfile.js]: can we switch .then here to await? codes really hard to read

File: server/playground/add-field-playground.js
Comment (arfpog) [inline server/playground/add-field-playground.js:1]: why is this here

File: client/src/components/SchoolQuerySearch.js (hunk 2 above)
Comment (arfpog) [reply client/src/components/SchoolQuerySearch.js]: ^^

**Review by Lewin-B — APPROVED** [review]
Looks good to me

**Review by arfpog — APPROVED** [review]
looks much much better now. 5/5

**Comment by aidanouckama** [comment]
Testing something.
